src/redisearch_rs/trie_rs/src/trie.rs

```diff
@@ -0,0 +1,917 @@
+            trie.mem_usage(),
+            1 * NODE_SIZE, // Unoccupied space for 1 node
+            "The computed size of an empty trie should equal the size of a single node"
+        );
+
+        let hello = b"hello".c_chars();
+        let world = b"world".c_chars();
+        let he = b"he".c_chars();
+        trie.insert(&hello, 1);
+
+        assert_eq!(
+            trie.mem_usage(),
+            1 * NODE_SIZE // Root node
+                + 10 // Label b"hello":  header (4) + padding(0) + capacity (5 * 1) + padding (1)
+                + 0, // 0 children
+            "The size of the trie should equal the size of 1 `Node<i32>`, 0 `ChildRef<i32>`s,\
+                and 1 label with a capacity of 5 elements and aligned to 2 bytes"
+        );
+
+        trie.insert(&world, 2);
+
+        assert_eq!(
+            trie.mem_usage(),
+            NODE_SIZE // Root node
+                + 10 // Label b"hello": header (4) + padding(0) + capacity (5 * 1) + padding (1)
+                + 10 // Label b"world": header (4) + padding(0) + capacity (5 * 1) + padding (1)
+                + thin_vec_mem_usage_for_cap::<ChildRef<i32>>(2), // 2 children: header (4) + padding(4) + capacity (2 * 32) + padding (0)
+            "The size of the trie should equal the size of 1 `Node<i32>`, 2 `ChildRef<i32>`s,\
+            and 2 labels with a capacity of 5 elements each and aligned to 2 bytes"
+        );
+
+        trie.insert(&he, 3);
+
+        assert_eq!(
+            trie.mem_usage(),
+            NODE_SIZE // Root node
+            + thin_vec_mem_usage_for_cap::<ChildRef<i32>>(2) // 2 children
+            // Node with label b"he" was split, and therefore has and additional capacity of 3 elements
+            + 10 // Label b"he" + XXX: header (4) + padding(0) + capacity(5 * 1) + padding(1)
+            + 10 // Label b"world": header (4) + padding(0) + capacity(5 * 1) + padding(1)
+            + thin_vec_mem_usage_for_cap::<ChildRef<i32>>(1) // 1 grandchild
+            + 8, // Label b"llo": header (4) + padding(0) + capacity(3 * 1) + padding(1)
+            "The size of the trie should equal the size of 1 `Node<i32>`, 2 `ChildRef<i32>`s for the
+            root's children, 1 `ChildRef<i32>` for the root's grandchild, \
+            and 3 labels with a capacity of 5 elements each and aligned to 2 bytes"
+        );
+
+        trie.remove(&he);
+
+        assert_eq!(
+            trie.mem_usage(),
+            NODE_SIZE // root node
+            + thin_vec_mem_usage_for_cap::<ChildRef<i32>>(2) // 2 children
+            + 10 // Label b"hello": header (4) + padding(0) + capacity (5 * 1) + padding (1)
+            + 10, // Label b"world": header (4) + padding(0) + capacity(5 * 1) + padding(1)
+            "The size of the trie should equal the size of 1 `Node<i32>`, 1 `ChildRef<i32>` for the
+            root's child, and 1 label with a capacity of 5 elements and aligned to 2 bytes"
+        );
+    }
+
+    #[derive(proptest_derive::Arbitrary, Debug)]
```

Comment:
Where is the definition of the flow that runs here? How many operations are there, and how are they generated?